hail/src/main/scala/is/hail/backend/service/ServiceBackend.scala

```diff
@@ -51,16 +125,84 @@ class ServiceBackend() extends Backend {
 
   def defaultParallelism: Int = 10
 
-  def broadcast[T: ClassTag](_value: T): BroadcastValue[T] = new BroadcastValue[T] {
+  def broadcast[T: ClassTag](_value: T): BroadcastValue[T] = new BroadcastValue[T] with Serializable {
     def value: T = _value
   }
```

Comment:
I don't think `BroadcastValue` should be serializable itself, right?

Reply:
It needs to be.

Right now, I'm just serializing broadcast values.  The idiom is:

```
  val valueBc = backend.broadcast(value)
  ...  { x => (fx, valueBc) } ...
```

where the closure is executed on the workers.  So the broadcast needs to be serializable.

In the future, it will be serializable, but the deserializer will be custom and it will load the through a broadcast mechanism (Google storage or some distributed memory store or whatever).

Since we currently storing the serialized function Google Storage and running 1 core per batch job, there's actually no benefit to a broadcast mechanism at the moment.